lib/compat/wordpress-6.7/class-gutenberg-rest-posts-controller-6-7.php

```diff
@@ -0,0 +1,699 @@
+			'search'         => 's',
+			'search_columns' => 'search_columns',
+			'slug'           => 'post_name__in',
+			'status'         => 'post_status',
+		);
+
+		/*
+		 * For each known parameter which is both registered and present in the request,
+		 * set the parameter's value on the query $args.
+		 */
+		foreach ( $parameter_mappings as $api_param => $wp_param ) {
+			if ( isset( $registered[ $api_param ], $request[ $api_param ] ) ) {
+				$args[ $wp_param ] = $request[ $api_param ];
+			}
+		}
+
+		// Check for & assign any parameters which require special handling or setting.
+		$args['date_query'] = array();
+
+		if ( isset( $registered['before'], $request['before'] ) ) {
+			$args['date_query'][] = array(
+				'before' => $request['before'],
+				'column' => 'post_date',
+			);
+		}
+
+		if ( isset( $registered['modified_before'], $request['modified_before'] ) ) {
+			$args['date_query'][] = array(
+				'before' => $request['modified_before'],
+				'column' => 'post_modified',
+			);
+		}
+
+		if ( isset( $registered['after'], $request['after'] ) ) {
+			$args['date_query'][] = array(
+				'after'  => $request['after'],
+				'column' => 'post_date',
+			);
+		}
+
+		if ( isset( $registered['modified_after'], $request['modified_after'] ) ) {
+			$args['date_query'][] = array(
+				'after'  => $request['modified_after'],
+				'column' => 'post_modified',
+			);
+		}
+
+		// Ensure our per_page parameter overrides any provided posts_per_page filter.
+		if ( isset( $registered['per_page'] ) ) {
+			$args['posts_per_page'] = $request['per_page'];
+		}
+
+		if ( isset( $registered['sticky'], $request['sticky'] ) ) {
+			$sticky_posts = get_option( 'sticky_posts', array() );
+			if ( ! is_array( $sticky_posts ) ) {
+				$sticky_posts = array();
+			}
+			if ( $request['sticky'] ) {
+				/*
+				 * As post__in will be used to only get sticky posts,
+				 * we have to support the case where post__in was already
+				 * specified.
+				 */
+				$args['post__in'] = $args['post__in'] ? array_intersect( $sticky_posts, $args['post__in'] ) : $sticky_posts;
+
+				/*
+				 * If we intersected, but there are no post IDs in common,
+				 * WP_Query won't return "no posts" for post__in = array()
+				 * so we have to fake it a bit.
+				 */
+				if ( ! $args['post__in'] ) {
+					$args['post__in'] = array( 0 );
+				}
+			} elseif ( $sticky_posts ) {
+				/*
+				 * As post___not_in will be used to only get posts that
+				 * are not sticky, we have to support the case where post__not_in
+				 * was already specified.
+				 */
+				$args['post__not_in'] = array_merge( $args['post__not_in'], $sticky_posts );
+			}
+		}
+
+		$args = $this->prepare_tax_query( $args, $request );
+
+		if ( ! empty( $request['format'] ) ) {
+			// If format is not an array, convert it to an array so that the
+			// required prefix can be added to all items.
+			$formats = is_array( $request['format'] ) ? $request['format'] : array( $request['format'] );
```

Comment:
could we think of a positive match to convert into an array here? is `$request['format'] === 'database'` a valid input? how could someone look at this line + comment and know what the data type is of a non-array input? how could it be analyzed for correctness or soundness?

```php
$formats = $request['format'];
if ( is_string( $formats ) ) {
	$formats = array( $formats );
}
if ( ! is_array( $formats ) ) {
	// not a valid input, abort processing formats.
}
```

Reply:
I don't understand what you mean.

Reply:
Maybe  @Mamaduka can help translate.

Reply:
IIRC, `format` here will always be an array; the schema ensures it. When an invalid parameter is provided, REST API will return `400 Bad Request - Invalid parameter(s): format`.

This conversion here is unnecessary.

Reply:
Yep exactly. And if a user passes just a single string, the REST API schema sanitization will take care of making it an array.